### PR TITLE
remove trace drivers' dependency on HttpTracerImpl

### PR DESCRIPTION
**Pull request by wbpcode**
Signed-off-by: wbpcode <comems@msn.com>

This PR is part of #16049 to support general tracing. Please check #16049 get more details.

Commit Message: remove trace drivers' dependency on HttpTracerImpl
Additional Description:

Now all tracers (zipkin, skywalking, etc.) will depend on HttpTracerImpl, making it difficult for Tracers to be reused by other protocols (Dubbo, Thrift, etc.). The purpose of this PR is to change this dependency.

Risk Level: Low (No new logic).
Testing: Add.
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by wbpcode** [comment]
Here is first sub PR. @lizan 

There are some points my be can help reviewers:
* This PR move `Tracing::Driver`, `Tracing::Config`, `Tracing::Span` etc to `trace_driver.h` as common interface for tracing.
* This PR separates Tracing::NullSpanImpl and some common values from `http_tracer_impl.h`.

File: source/common/tracing/common_values.h [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+namespace Envoy {
+namespace Tracing {
+
+/**
+ * Tracing tag names.
+ */
+class TracingTagValues {
+public:
+  // OpenTracing standard tag names.
+  const std::string Component = "component";
```
Comment (dio) [inline source/common/tracing/common_values.h:16]: Not sure if we can make all of these as constexpr. WDYT?

Comment (wbpcode) [reply source/common/tracing/common_values.h:16]: These values are separated from http_tracer_impl.h derectly. I prefer to add a TODO, and then complete the const to constexpr migration in a new PR.

Comment (lizan) [reply source/common/tracing/common_values.h:16]: sgtm

**Review by lizan — APPROVED** [review]

**Pull request merged**